### ISSUE TITLE
Change Headers of GraphQlReplicationState

### ISSUE THREAD

**Issue by DDoerner** [pubkey/rxdb#2389]
## Case
Feature discussion

## Issue
Similar to issue  #1977, we have the case that bearer tokens for an active gql replication need to be changed, which at the moment requires to restart the whole replication. Instead I'd suggest to add a method `setHeaders()` to the replication state object. The change should be minimal and I'd be happy to write a PR, but wanted to discuss the suggestion first.

## Code

```js
setHeaders(headers: { [k: string]: string }): void {
  this.client = GraphQLClient({
    url: this.url,
    headers
  });
}
```

**Comment by pubkey** [comment]
Sorry for the late reply.
I am not sure if the proposed change is the correct solution for the problem.
You swap out the complete graphql client, not just the headers. What happens with ongoing websockets from the old client?
Also the client is public atm so I do not see the need for maintaining `setHeaders` when the user can switch out the client anyway.

**Comment by DDoerner** [comment]
As far as I can tell from the repo of the graphql-client, it doesn't use websockets at all and has no real internal state (https://github.com/nordsimon/graphql-client/blob/master/index.js). So replacing it shouldn't be a problem, especially since it doesn't offer any methods for changing the url or header of the client object.

We could simply switch the client ourselves, that's true, but it feels a bit hacky since it requires inside knowledge. This is why I think that a proper method would be the cleaner solution.

**Comment by pubkey** [comment]
Ah yes you are right. I think we should merge this. Can you add at least one test to ensure it is not broken in future refactorings or version upgrades?

**Comment by DDoerner** [comment]
Will do, I'll try to think of a test that makes sense.

**Comment by pubkey** [comment]
fixed in #2399
